scrapi/base/helpers.py

```diff
@@ -135,7 +136,10 @@ def oai_process_uris(*args):
                 provider_uris.append(found_url)
 
     try:
-        canonical_uri = (provider_uris + object_uris)[0]
+        if use_doi and len(object_uris) > 0:
+            canonical_uri = object_uris[0]
```

Comment:
Also, do harvesters that use this flag not have any providerUris? The canonicalUri would ideally be some sort of splash page